pkg/v1/tkg/clusterclient/clusterclient.go

```diff
@@ -634,7 +634,12 @@ func (c *client) WaitForAutoscalerDeployment(deploymentName, namespace string) e
 }
 
 func (c *client) WaitForAVIResourceCleanUp(statefulSetName, namespace string) error {
-	return c.GetResource(&appsv1.StatefulSet{}, statefulSetName, namespace, VerifyAVIResourceCleanupFinished, &PollOptions{Interval: CheckResourceInterval, Timeout: AVIResourceCleanupTimeout})
+	err := c.GetResource(&appsv1.StatefulSet{}, statefulSetName, namespace, VerifyAVIResourceCleanupFinished, &PollOptions{Interval: CheckResourceInterval, Timeout: AVIResourceCleanupTimeout})
+	// retry once when network condition is poor
+	if apierrors.IsServiceUnavailable(err) {
+		return c.GetResource(&appsv1.StatefulSet{}, statefulSetName, namespace, VerifyAVIResourceCleanupFinished, &PollOptions{Interval: CheckResourceInterval, Timeout: AVIResourceCleanupTimeout})
```

Comment:
do we have something better like `Eventually` to do this?

Reply:
+1
Retrying once may not be helpful as the next attempt will happen within milliseconds.

Reply:
I believe that `GetResource` function is already did what you suggest here, it will keep trying until time out

Reply:
Got it